src/java/org/apache/cassandra/index/sai/disk/v2/V2VectorIndexSearcher.java

```diff
@@ -291,27 +295,18 @@ private CloseableIterator<RowIdWithScore> orderByBruteForce(VectorFloat<?> query
      */
     private CloseableIterator<RowIdWithScore> orderByBruteForce(CompressedVectors cv,
                                                                 VectorFloat<?> queryVector,
-                                                                IntArrayList segmentRowIds,
+                                                                Int2IntHashMap segmentToOrdinalMap,
                                                                 int limit,
                                                                 int rerankK) throws IOException
     {
-        var approximateScores = new ArrayList<BruteForceRowIdIterator.RowWithApproximateScore>(segmentRowIds.size());
+        var approximateScores = new ArrayList<BruteForceRowIdIterator.RowWithApproximateScore>(segmentToOrdinalMap.size());
         var similarityFunction = indexContext.getIndexWriterConfig().getSimilarityFunction();
         var scoreFunction = cv.precomputedScoreFunctionFor(queryVector, similarityFunction);
 
-        try (var ordinalsView = graph.getOrdinalsView())
-        {
-            for (int i = 0; i < segmentRowIds.size(); i++)
-            {
-                int segmentRowId = segmentRowIds.getInt(i);
-                int ordinal = ordinalsView.getOrdinalForRowId(segmentRowId);
-                if (ordinal < 0)
-                    continue;
-
-                var score = scoreFunction.similarityTo(ordinal);
-                approximateScores.add(new BruteForceRowIdIterator.RowWithApproximateScore(segmentRowId, ordinal, score));
-            }
-        }
+        segmentToOrdinalMap.forEach((segmentRowId, ordinal) -> {
```

Comment:
I see we only ever iterator over this map in its entirety. Makes me wonder if we need the Int2Int map or if a `Collection<RowOrdinalPair>` would make sense. Either will be fast, so I am going to leave it as is.

Reply:
the nice thing about int2int is, it doesn't box

(but you need to use .forEachInt here to get part of the benefit)

Reply:
Thanks for pointing that out, I missed that nuance. In looking at the impact of load factor, I decided that it'd be straight forward to implement our own `IntIntPairArray`, so I did that in the next commit. No boxing and no resizing needed. Probably an unnecessary optimization, but it's very straight forward and completely tested.